src/components/Attachments/AttachmentsReducer.js

```diff
@@ -0,0 +1,107 @@
+import { handleActions } from 'redux-actions';
+import dotProp from 'dot-prop-immutable';
+import * as Actions from './AttachmentsActions';
+
+/**
+ * Use it for debug when styling
+ */
+const debug = false;
+
+const debugDefaultState = {
+  uploads: [
+    {
+      file: {
+        name: 'File uploaded',
+      },
+      status: 'done',
+      result: {
+        id: 'bla bla',
+      },
+    },
+    {
+      file: {
+        name: 'File uploading',
+      },
+      status: 'uploading',
+      progress: 0.47,
+    },
+    {
+      file: {
+        name: 'File error',
+      },
+      status: 'error',
+      error: {
+        message: 'Something went wrong',
+      },
+    },
+    {
+      file: {
+        name: 'File initializing',
+      },
+    },
+  ],
+};
+
+const normalDefaultState = {
+  uploads: [],
+};
+
+const defaultState = debug ? debugDefaultState : normalDefaultState;
+
+function updateFile(updates) {
+  return (state, { payload: { file, uploadFolderId, ...payload } }) =>
+    dotProp.set(state, 'uploads', uploads =>
+      uploads.map(item => {
+        if (!(item.file === file && item.uploadFolderId === uploadFolderId)) {
+          return item;
+        }
+
+        if (typeof updates === 'function') {
+          return {
+            file,
+            uploadFolderId,
+            ...updates(payload),
+          };
+        }
+
+        return {
+          file,
+          uploadFolderId,
+          ...updates,
+        };
+      }),
+    );
+}
+
+export const attachmentsReducer = handleActions(
+  {
+    [Actions.addUploadingFiles]: (
+      state,
+      { payload: { files, uploadFolderId } },
+    ) =>
+      dotProp.set(state, 'uploads', uploads => [
+        ...Array.from(files).map(file => ({ file, uploadFolderId })),
+        ...uploads,
+      ]),
+    [Actions.fileUploadInitialized]: updateFile({ status: 'initialized' }),
+    [Actions.fileUploadGetLocation]: updateFile({ status: 'getLocation ' }),
```

Comment:
`'getLocation '` has a space in the end. So is going to fail for `if (status === 'getLocation')` ?
Maybe would make sense to use constants (a kind of ENUM) for all the status stings? :)

Reply:
And what does the `getLocation` status mean? How can status be `getLocation`? Maybe `gettingLocation`? Or `gotLocation` ? And which location are we talking about?
I see it's coming from the ResumableUpload lib... But could you please add a comment of what the hell is that, why are we using it? :)

Reply:
>> I see it's coming from the ResumableUpload lib...

right

constants, surely, are a good idea, you are right

but I thought to defer it for the future for next reason:

They are coming from `third-party` code which I would like to refactor and restructure, I didn't want to do it now to save development time reasons

I'll add constants with comments in next push